extension/webviews/components/copilot/CopilotInputForm.svelte

```diff
@@ -0,0 +1,115 @@
+<script lang="ts">
+  import { onDestroy, onMount, tick } from 'svelte';
+  import Button from '../Button.svelte';
+  import { copilotMessages, copilotReferToCodeFile, copilotReferToDoclinThreads } from '../../state/store';
+  import { CopilotRole } from '../../enums';
+
+  let input: string = '';
+  let disableSubmit: boolean = false;
+  let textarea: HTMLTextAreaElement;
+
+  $: autoResizeTextarea(), input;
+
+  onMount(async () => {
+    window.addEventListener('message', messageEventListener);
+  });
+
+  onDestroy(() => {
+    window.removeEventListener('message', messageEventListener);
+  });
+
+  const messageEventListener = async (event: Event) => {
+    const type = event.data.type;
+    const value = event.data.value;
+    switch (type) {
+      case 'postCopilotPrompt':
+        const reply = value;
+        $copilotMessages = [
+          ...$copilotMessages.slice(0, $copilotMessages.length - 1),
+          { role: CopilotRole.ASSISTANT, content: reply },
+        ];
+        disableSubmit = false;
+        break;
+    }
+  };
+
+  const handleKeyDown = (event: KeyboardEvent): void => {
+    if (event.key === 'Enter' && !event.shiftKey) {
+      event.preventDefault();
+      submitPrompt();
+    } else if (event.key === 'ArrowUp') {
+      retrievePreviousPrompt();
+    }
+  };
+
+  const submitPrompt = async () => {
+    if (disableSubmit || !input) {
+      return;
+    }
+
+    disableSubmit = true;
+
+    $copilotMessages = [
+      ...$copilotMessages,
+      { role: CopilotRole.USER, content: input },
+      { role: CopilotRole.ASSISTANT, content: 'typing...' },
```

Comment:
we shouldn't need this at the component level. It should be at the api level/copilotPrompt. Frontend should only send the user input.

Reply:
user messages are not stored in the database. They are only stored locally in store.ts